From 139397e64e17241f74dad09ea850c1edbb75d803 Mon Sep 17 00:00:00 2001
From: Samuel Holland <samuel@sholland.org>
Date: Mon, 22 Jun 2020 19:02:43 -0500
Subject: [PATCH] sun50i: a64: pinephone: Work around ATF bug

Mainline ATF turns on DRIVEVBUS, which turns off VBUS, which causes the
phone to lose power and reset if the battery is dead. Work around this
by removing the property from U-Boot's copy of the DTS.

Signed-off-by: Samuel Holland <samuel@sholland.org>
---
 arch/arm/dts/sun50i-a64-pinephone-1.0.dts | 4 ----
 arch/arm/dts/sun50i-a64-pinephone-1.1.dts | 4 ----
 2 files changed, 8 deletions(-)

diff --git a/arch/arm/dts/sun50i-a64-pinephone-1.0.dts b/arch/arm/dts/sun50i-a64-pinephone-1.0.dts
index 77f6869f85f..5ac8d19d4c4 100644
--- a/arch/arm/dts/sun50i-a64-pinephone-1.0.dts
+++ b/arch/arm/dts/sun50i-a64-pinephone-1.0.dts
@@ -10,10 +10,6 @@
 	compatible = "pine64,pinephone-1.0", "allwinner,sun50i-a64";
 };
 
-&axp803 {
-	x-powers,drive-vbus-en;
-};
-
 /*
  * The N_VBUSEN pin is disconnected, but we need to inform the PMIC about
  * the VBUS status anyway. To avoid the pin from floating and to inform
diff --git a/arch/arm/dts/sun50i-a64-pinephone-1.1.dts b/arch/arm/dts/sun50i-a64-pinephone-1.1.dts
index 863e158705d..7ebac3ab9b6 100644
--- a/arch/arm/dts/sun50i-a64-pinephone-1.1.dts
+++ b/arch/arm/dts/sun50i-a64-pinephone-1.1.dts
@@ -10,10 +10,6 @@
 	compatible = "pine64,pinephone-1.1", "allwinner,sun50i-a64";
 };
 
-&axp803 {
-	x-powers,drive-vbus-en;
-};
-
 &backlight {
 	power-supply = <&reg_ldo_io0>;
 	/*
